jpo-ode-plugins/src/main/java/us/dot/its/jpo/ode/plugin/j2735/builders/TravelerMessageFromHumanToAsnConverter.java

```diff
@@ -40,1080 +41,1114 @@
+      Long transformedLon = LatitudeBuilder.j2735Latitude(lonOffset);
+      Long transformedLat = LongitudeBuilder.j2735Longitude(latOffset);
+      ObjectNode latLong = JsonUtils.newNode().put(LON, transformedLon).put(LAT, transformedLat);
+      if (deltaText.equals(NODE_XY)) {
+        innerNode.set(nodeOffsetPointLL(transformedLat, transformedLon), latLong);
       } else {
-         throw new IllegalArgumentException("Invalid node X/Y offset: " + transformedX + "/" + transformedY
-               + ". Values must be between a range of -327.68/+327.67 meters.");
+        innerNode.set(deltaText, latLong);
       }
-   }
+    }
+
+    deltaNode.set(DELTA, innerNode);
+
+    return deltaNode;
+
+  }
+
+  // NodeOffsetPointXY contains one of:
+  // node-XY1 Node-XY-20b, -- node is within 5.11m of last node
+  // node-XY2 Node-XY-22b, -- node is within 10.23m of last node
+  // node-XY3 Node-XY-24b, -- node is within 20.47m of last node
+  // node-XY4 Node-XY-26b, -- node is within 40.96m of last node
+  // node-XY5 Node-XY-28b, -- node is within 81.91m of last node
+  // node-XY6 Node-XY-32b, -- node is within 327.67m of last node
+  // node-LatLon Node-LLmD-64b, -- node is a full 32b Lat/Lon range
+  private static String nodeOffsetPointXY(long transformedX, long transformedY) {
+    long transformed = Math.abs(transformedX) | Math.abs(transformedY);
+    if ((transformed & (-1 << 10)) == 0) {
+      return "node-XY1";
+    } else if ((transformed & (-1 << 11)) == 0) {
+      return "node-XY2";
+    } else if ((transformed & (-1 << 12)) == 0) {
+      return "node-XY3";
+    } else if ((transformed & (-1 << 13)) == 0) {
+      return "node-XY4";
+    } else if ((transformed & (-1 << 14)) == 0) {
+      return "node-XY5";
+    } else if ((transformed & (-1 << 16)) == 0) {
+      return "node-XY6";
+    } else {
+      throw new IllegalArgumentException(
+          "Invalid node X/Y offset: " + transformedX + "/" + transformedY
+              + ". Values must be between a range of -327.68/+327.67 meters.");
+    }
+  }
+
+  /**
+   * Ensures compliance with the J2735 2024 standard by checking for old fields in the given data frame.
+   *
+   * @param dataFrame the JSON object representing the data frame to be checked
+   * @throws IllegalArgumentException if any old fields are found
+   */
+  public static void ensureComplianceWithJ2735Revision2024(ObjectNode dataFrame) {
+    // Check and throw exception if old fields are found
+    if (dataFrame.has(SSP_MSG_CONTENT) || dataFrame.has(SSP_MSG_TYPES) ||
+        dataFrame.has(SSP_TIM_RIGHTS) || dataFrame.has(SSP_LOCATION_RIGHTS) ||
+        dataFrame.has(SSP_MSG_RIGHTS_1) || dataFrame.has(SSP_MSG_RIGHTS_2) ||
+        dataFrame.has(NOT_USED) || dataFrame.has(NOT_USED_1) ||
+        dataFrame.has(NOT_USED_2) || dataFrame.has(NOT_USED_3) ||
+        dataFrame.has(DURATON_TIME_MISSPELLED)) {
+      throw new IllegalArgumentException(

```

Comment:
Good call, and thanks for the links! I've updated TravelerMessageFromHumanToAsnConverter to throw a custom checked  NoncompliantFieldsException instead of the runtime IllegalArgumentException.